src/main/kotlin/io/emeraldpay/dshackle/config/reload/ReloadConfigSetup.kt

```diff
@@ -0,0 +1,142 @@
+package io.emeraldpay.dshackle.config.reload
+
+import io.emeraldpay.dshackle.Chain
+import io.emeraldpay.dshackle.Global.Companion.chainById
+import io.emeraldpay.dshackle.config.UpstreamsConfig
+import io.emeraldpay.dshackle.foundation.ChainOptions
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import sun.misc.Signal
+import sun.misc.SignalHandler
+import java.util.stream.Collectors
+
+@Component
+class ReloadConfigSetup(
+    private val reloadConfigService: ReloadConfigService,
+    private val reloadConfigUpstreamService: ReloadConfigUpstreamService,
+) : SignalHandler {
+
+    companion object {
+        private val log = LoggerFactory.getLogger(this::class.java)
+        private val signalHup = Signal("HUP")
+    }
+
+    init {
+        Signal.handle(signalHup, this)
+    }
+
+    override fun handle(sig: Signal) {
+        if (sig == signalHup) {
+            try {
+                log.info("Reloading config...")
+
+                reloadConfig()
+
+                log.info("Config is reloaded")
+            } catch (e: Exception) {
+                log.warn("Config is not reloaded, cause - ${e.message}")
```

Comment:
maybe log exception as last param? With stacktrace, because sometimes it can be some generic parse error and we need to figure out it origin